app/static/main.js

```diff
@@ -19,20 +19,43 @@ function getStatus(taskID) {
   })
   .then(response => response.json())
   .then(res => {
+    const taskStatus = res.task_status;
+    const taskResult = res.task_result;
+    
     const html = `
       <tr>
         <td>${taskID}</td>
-        <td>${res.task_status}</td>
-        <td>${res.task_result}</td>
+        <td>${taskStatus}</td>
+        <td>${taskResult}</td>
       </tr>`;
     const newRow = document.getElementById('tasks').insertRow(0);
     newRow.innerHTML = html;
 
-    const taskStatus = res.task_status;
+    
+    
+    if (taskStatus === 'SUCCESS') {
+      const downloadURL = res.task_result;
+      const downloadButton = document.getElementById('dialog-box');
+      downloadButton.href = downloadURL;
+      showDialog('dialog-box');
+    }
     if (taskStatus === 'SUCCESS' || taskStatus === 'FAILURE') return false;
     setTimeout(function() {
       getStatus(res.task_id);
     }, 1000);
+    
   })
   .catch(err => console.log(err));
 }
+
+function showDialog(x) {
+	// show the dialog
+	var dialog = document.getElementById(x);
+	dialog.showModal();
+}
+
+function closeDialog(x) {
+	// show the dialog
```

Comment:
is this comment right?

Reply:
Yes, line 58 should ready
`// hide the dialog`
Will update